papers/source/d1664.bs

```diff
@@ -130,22 +130,28 @@ In short, this paper formalizes the work done in P1739 by giving it an expositio
 
 # Design # {#design}
 
-The concept is simple and is given in 2 exposition-only types added to the standard:
+The design is simple and is given in 2 exposition-only concepts added to the standard:
 
 ```
 template <typename R>
-concept pair-reconstructible-range = 
-    forwarding-range<std::remove_reference_t<R>> && 
-    std::Constructible<std::remove_reference_t<R>, iterator_t<R>, sentinel_t<R>>;
+concept pair-reconstructible-range =
+    Range<R> &&
+    forwarding-range<std::remove_reference_t<R>> &&
+    std::Constructible<R, iterator_t<R>, sentinel_t<R>>;
 
 template <typename R>
-concept range-reconstructible-range = 
-    forwarding-range<std::remove_reference_t<R>> && 
-    std::Constructible<std::remove_reference_t<R>, std::ranges::subrange<iterator_t<R>, sentinel_t<R>>>;
+concept range-reconstructible-range =
+    Range<R> &&
+    forwarding-range<std::remove_reference_t<R>> &&
+    std::Constructible<R, std::ranges::subrange<iterator_t<R>, sentinel_t<R>>>;
 ```
 
-It is the formalization that a range can be constructed from its begin iterator and end iterator/sentinel. It also provides an exposition concept for allowing a range to be constructed from a `subrange` of its iterator/sentinel pair. This allows a developer to propagate the input type's properties after modifying its iterators for some underlying work, algorithm or other effect. This concept is also the basis of the idea behind [[p1739r0]].
+It is the formalization that a range can be constructed from its begin iterator and end iterator/sentinel. It also provides an exposition-only concept for allowing a range to be constructed from a `subrange` of its iterator/sentinel pair. This allows a developer to propagate the input type's properties after modifying its iterators for some underlying work, algorithm or other effect. This concept is also the basis of the idea behind [[p1739r0]].
 
+Both concepts require that the type with any references removed model the exposition-only concept `forwarding-range`.
+This ensures that the validity of the iterators is in fact independent of the lifetime of the range they originate from
+and that a "reconstructed" range does not depend on the original.
+We remove reference before performing this check, because all reference types that model `Range` also model `forwarding-range` and the intent of the proposed changes is narrower: (re)construction is assumed to be in constant time (this typically implies that `R` also models `View`, but it is sufficient to check `forwarding-range<std::remove_reference_t<R>>`). Note that this explicitly excludes types like `std::vector<int> const &` from being reconstructible.

```

Comment:
I tried to explain my reasoning inside the proposal, because other people will likely have the same questions. What do you think about the matter?

Reply:
This is helpful. I am new to ranges and have only been using them for a short time: I appreciate the help in getting it right.

Reply:
No worries, glad I could convince you. I love ranges, but some things just take really long to wrap your head around... and naming... well try to explain to people "ForwardRange" vs "forwarding-range" 🤦‍♂